Sources/TuistGenerator/Generator/DerivedFileGenerator.swift

```diff
@@ -77,13 +83,19 @@ final class DerivedFileGenerator: DerivedFileGenerating {
         // Generate the Info.plist
         try targetsWithGeneratableInfoPlists.forEach { target in
             guard let infoPlist = target.infoPlist else { return }
-            guard case let InfoPlist.dictionary(dictionary) = infoPlist else { return }
+            var dictionary: [String: Any]?
+
+            if case let InfoPlist.dictionary(_dictionary) = infoPlist {
+                dictionary = _dictionary.mapValues { $0.value }
+            } else if case let InfoPlist.extendingDefault(extended) = infoPlist {
+                dictionary = self.infoPlistContentProvider.content(target: target, extendedWith: extended)
+            }
+            if dictionary == nil { return }
 
             let path = DerivedFileGenerator.infoPlistPath(target: target, sourceRootPath: sourceRootPath)
             if fileHandler.exists(path) { try fileHandler.delete(path) }
 
-            let outputDictionary = dictionary.mapValues { $0.value }
-            let data = try PropertyListSerialization.data(fromPropertyList: outputDictionary,
+            let data = try PropertyListSerialization.data(fromPropertyList: dictionary!,
```

Comment:
the IUO can be omitted if you ensure the value is available by the time this code is executed. e.g.

```swift
let dictionary: [String: Any]

if case let InfoPlist.dictionary(_dictionary) = infoPlist {
    dictionary = _dictionary.mapValues { $0.value }
} else if case let InfoPlist.extendingDefault(extended) = infoPlist {
    dictionary = self.infoPlistContentProvider.content(target: target, extendedWith: extended)
} else { 
    return 
}

```

Reply:
Agree on this one - to help, the logic could be extracted to private helper method too :)